frontend/src/components/Shared/CrossingMapPage/CrossingMapSidebar.js

```diff
@@ -70,6 +72,35 @@ class CrossingMapSidebar extends Component {
     }
   };
 
+  getNearbyCrossings = () => {
+    const {
+      openCrossings,
+      closedCrossings,
+      cautionCrossings,
+      longtermCrossings,
+      showOpen,
+      showClosed,
+      showCaution,
+      showLongterm,
+      center,
+    } = this.props;
+
+    let nearbyCrossings = [];
+
+    if (showOpen && openCrossings) nearbyCrossings.push(...openCrossings);
+    if (showClosed && closedCrossings) nearbyCrossings.push(...closedCrossings);
+    if (showCaution && cautionCrossings)
+      nearbyCrossings.push(...cautionCrossings);
+    if (showLongterm && longtermCrossings)
+      nearbyCrossings.push(...longtermCrossings);
+
+    return nearbyCrossings.length
+      ? _.sortBy(nearbyCrossings, c =>
+          geolib.getDistance(center, JSON.parse(c.geojson).coordinates),
+        ).slice(0, 20)
```

Comment:
Should this be `0, 21`? Could use `_.take` instead. https://lodash.com/docs#take

Reply:
I honestly just picked 20 to get it so we weren't rendering every single nearby crossing. I think the best option for the future would be to use react-virtualized for infinite scrolling as we do with the crossing list view. Made an issue https://github.com/cityofaustin/ctxfloods/issues/156